lib/decorators/index.ts

```diff
@@ -1,12 +1,30 @@
+/**
+ * 2021-04-07 by MrSpoocy: Fix issue if you use ES6 or higher (https://github.com/vuex-orm/vuex-orm/commit/25b458bd4de72f64ad4e8b3a3983ace062be8cfd)
+ *
+ * @param {string} aggregateType
+ * @param eventHandlersType
+ * @constructor
+ */
 export function Aggregate(aggregateType: string, eventHandlersType) {
-  return function handlesEventsFor<T extends { new(...args: any[]): {} }>(
-      constructor: T
-  ) {
-    return class extends constructor {
-      aggregateType = aggregateType;
-      initialState = eventHandlersType.prototype.initialState;
-      eventHandlers = eventHandlersType.eventHandlers;
-    };
+  return function handlesEventsFor<T extends { new(...args: any[]): {} }>(constructor: T) {
+    let proxy: T
+    try {
+      proxy = new Function('constructor', 'aggregateType', 'eventHandlersType', `
+        'use strict';
+        return class extends constructor {
+            aggregateType = aggregateType;
+            initialState = eventHandlersType.prototype.initialState;
+            eventHandlers = eventHandlersType.eventHandlers;
+        };
+    `)(constructor, aggregateType, eventHandlersType);
+    } catch {
```

Comment:
In what scenarios will an exception be thrown in a Node environment? Is it only a browser-related issue maybe we can skip the try/catch here?

Reply:
Not tested yet, I tested it at Node 15.12 and have make the fix "fast" for me that I can use the client. Have not check in the deep what I really need or what can remove. I trust in you to figure out :)

Reply:
Sounds good. Will make a new release with this change and we'll look deeper into it.